app/models/user.rb

```diff
@@ -1,8 +1,11 @@
 # frozen_string_literal: true
 
 class User < ApplicationRecord
+  after_save_commit :create_shopping_cart
```

Comment:
zaufaj mi, ze nie chcesz uzywac callbackow :)

Reply:
No właśnie czytałem że to winner w codesmell. :) Przerzuciłem create do registration controllera dla devise.